src/wp-includes/block-bindings/sources/post-meta.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * Add the post_meta source to the Block Bindings API.
+ *
+ * @since 6.5.0
+ * @package WordPress
+ */
+function post_meta_source_callback( $source_attrs ) {
+	if ( ! isset( $source_attrs['key'] ) ) {
+		return null;
+	}
+
+	// Use the postId attribute if available
+	if ( isset( $source_attrs['postId'] ) ) {
+		$post_id = $source_attrs['postId'];
+	} else {
+		// $block_instance->context['postId'] is not available in the Image block.
```

Comment:
Can we track this in an issue in Gutenberg and figure out why the context is not always filled. (like in the frontend)

Reply:
Sure, I'll do that. I wanted to triage it a bit more to ensure it was a real problem and not something I was doing wrong. I'll open the issue after that.

Reply:
Oh, I just realized that it is working in the editor because we created a filter to add the context: [link](https://github.com/WordPress/gutenberg/blob/ece4d7d16b5964fa92336e3b79631e76ee88ddd4/packages/block-editor/src/hooks/use-bindings-attributes.js#L127-L145). Maybe we need something similar in PHP.

Reply:
Using something like this seems to solve the issue:

```
function gutenberg_add_context_for_block_bindings( $args, $name ) {
	$allowed_blocks = array(
		'core/paragraph' => array( 'content' ),
		'core/heading'   => array( 'content' ),
		'core/image'     => array( 'url', 'title', 'alt' ),
		'core/button'    => array( 'url', 'text' ),
	);

	if ( ! isset( $allowed_blocks[ $name ] ) ) {
		return $args;
	}
	if ( ! isset( $args['uses_context'] ) ) {
		$args['uses_context'] = array( 'postId', 'postType' );
	} else {
		// Check if 'postId' exists in 'uses_context' and add it if it doesn't
		if ( ! in_array( 'postId', $args['uses_context'], true ) ) {
			$args['uses_context'][] = 'postId';
		}
		// Check if 'postType' exists in 'uses_context' and add it if it doesn't
		if ( ! in_array( 'postType', $args['uses_context'], true ) ) {
			$args['uses_context'][] = 'postType';
		}
	}

	return $args;
}
add_filter( 'register_block_type_args', 'gutenberg_add_context_for_block_bindings', 10, 2 );
```

Furthermore, I think if we follow this approach we don't need [the editor hook](https://github.com/WordPress/gutenberg/blob/ece4d7d16b5964fa92336e3b79631e76ee88ddd4/packages/block-editor/src/hooks/use-bindings-attributes.js#L127-L145) we are using, right? Do you think we should pursue this approach?

Reply:
I ended up creating a pull request in Gutenberg in case we want to follow this path: [link](https://github.com/WordPress/gutenberg/pull/58554).